airflow/utils/sqlalchemy.py

```diff
@@ -590,7 +592,17 @@ def tuple_not_in_condition(
 
     :meta private:
     """
-    if settings.engine.dialect.name != "mssql":
+    warnings.warn(
```

Comment:
Or maybe better remove this function 🤔

Reply:
No need to deprecate it, this has `:meta private:` so no user code should depend on it. I feel it’s fine to keep this though, it’s a mirror to `tuple_in_condition` and might be useful at some point.

Reply:
I hope we also could get rid of `tuple_in_condition`.

Reply:
Anyway, I would remove deprecation note